app/controllers/batch_controller.rb

```diff
@@ -0,0 +1,64 @@
+require 'json'
+
+get "/batch/upload" do
+  erb :'batches/upload'
+end
+
+post "/batch/upload" do
+  #   # response = {manipulations: parse_bill, user_data: user_data}.to_json
+  bill_data = parse_bill.to_json
+  session[:batch_id] = Batch.create(bill_data: bill_data).id
```

Comment:
This is a bad use of session. Storing ids of resources in the session and then using them in later requests is a very bad practice. HTTP is largely a stateless protocol. Instead of storing this ID in the session pass the ID in the url of the requests that are currently using `session[:batch_id]`